[FLOC-4255] Add documentation for Flocker deploy on RHEL 7.2

myechuri opened the pull request:
Add documentation for installing flocker node and flocker cli on RHEL 7.2. Relevant sections start at http://clusterhq-staging-docs.s3-website-us-east-1.amazonaws.com/add-rhel-documentation-FLOC-4255/docker-integration/manual-install.html

Setting up RHEL CI environment will follow this PR.

davidoliverSP2 commented on docs/installation/install-node.rst:
```diff
@@ -44,6 +44,54 @@ If you do not have any nodes, the following guides will help you set some up, wi
+   
+   Run the following commands as root on the target node:
+
+   .. prompt:: bash root@rhel:~$
```
If this is run as root, should the prompt directive be `.. prompt:: bash [root@rhel]#`?

myechuri replied:
Thanks for pointing out! Fixed.

davidoliverSP2 commented on docs/installation/install-node.rst:
```diff
@@ -44,6 +44,54 @@ If you do not have any nodes, the following guides will help you set some up, wi
+   
+      yum install -y clusterhq-flocker-docker-plugin
+
+   .. XXX FLOC-3454 to create a task directive for installing the plugin
```
I added indents here (and below where the same comment was used), as it was causing the numbering to get misaligned.

myechuri replied:
Thanks!

davidoliverSP2 commented on docs/installation/enabling-control-service.rst:
```diff
@@ -29,6 +29,27 @@ For more details on configuring the firewall, see the `FirewallD documentation`_
+RHEL 7.2
+========
+
+.. prompt:: bash root@rhel:~$
```
The other instructions in this document use prompt directives using root - should this be `.. prompt:: bash [root@rhel]#?`

myechuri replied:
Fixed.

davidoliverSP2 commented:
Thanks @myechuri 

I added a few commits that cleaned up some indenting - I added an indent to a comment (which I don't think you added in this PR), and removed some indents from code blocks.

myechuri commented:
Thanks for the quick review, @davidoliverSP2 ! Addressed your review comments.

davidoliverSP2 commented on docs/docker-integration/install-node.rst:
```diff
@@ -21,6 +21,12 @@ Installing the Flocker Node Services
 * :ref:`aws-install-docker`
 * :ref:`rackspace-install-docker`
 
+.. _rhel-7-install-docker:
```
Does this need to be replicated in the Kubernetes and Standalone instructions? Currently the RHEL instructions for Installing the Node services are only appearing in the Docker integration instructions.

myechuri replied:
Thanks for pointing out, @davidoliverSP2 . Added.

davidoliverSP2 commented on docs/installation/install-client.rst:
```diff
@@ -57,6 +57,19 @@ On Ubuntu 14.04 (64-bit), the Flocker CLI can be installed from the ClusterHQ re
 
 .. begin-body-installing-client-linux
 
+Installing on RHEL 7.2
```
These instructions have been embedded into the `installing-client-linux` section - I'm not sure if this was intentional - it probably makes sense unless there is a scenario where we would want to publish instructions for "other linux distributions" but not RHEL. 

As it stands, every version of our install instructions (Docker/Kubernetes/Standalone) all have RHEL in the installing the client, but not all of them have RHEL in the node instructions (see my previous comment)

davidoliverSP2 commented:
Thanks @myechuri 

I've raised a few comments relating to whether the RHEL instructions need to appear for both installing the client and node services - currently they appear in all of the client instructions, but only the Docker integration for the node services.

I would suggest best practice is to label up the RHEL instructions in the `installation/install-client.rst` file as you have in the `installation/install-node.rst` document, then in each of the integration folders add an include directive. 

If you'd rather pass this over to me, I'm happy to pick it up - just move the Jira ticket to me!

myechuri commented:
> I would suggest best practice is to label up the RHEL instructions in the installation/install-client.rst file as you have in the installation/install-node.rst document, then in each of the integration folders add an include directive.

Thanks for suggesting, @davidoliverSP2 . Refactored client install instructions accordingly - definitely makes it cleaner.

myechuri commented:
@davidoliverSP2 : addressed your comments and verified that the staging results comply: http://clusterhq-staging-docs.s3-website-us-east-1.amazonaws.com/add-rhel-documentation-FLOC-4255

Please let me know if you have any further comments. Thanks!
